host-macros/src/lib.rs

```diff
@@ -144,25 +151,104 @@ pub fn gatt_service(args: TokenStream, item: TokenStream) -> TokenStream {
 }
 
 /// Check if a field has a characteristic attribute and parse it.
+///
+/// If so also check if that field has descriptors and/or docstrings.
+///
+/// # Example
+///
+/// ```rust
+/// use trouble_host::prelude::*;
+///
+/// #[gatt_service(uuid = "180f")]
+/// struct BatteryService {
+///    /// Docstrings can be
+///    /// Multiple lines
+///    #[characteristic(uuid = "2a19", read, write, notify, value = 99, on_read = battery_level_on_read)]
+///    #[descriptor(uuid = "2a20", read, write, notify)]
+///    #[descriptor(uuid = "2a20", read, write, notify, value = "Demo description")]
+///    level: u8,
+///    #[descriptor(uuid = "2a21", read, write, notify, value = VAL)]
+///    #[characteristic(uuid = "2a22", read, write, notify, value = 42.0)]
+///    /// Can be in any order
+///    rate_of_discharge: f32,
+///}
+/// ```
 fn check_for_characteristic(
     field: &syn::Field,
     err: &mut Option<syn::Error>,
     characteristics: &mut Vec<Characteristic>,
 ) -> bool {
     const RETAIN: bool = true;
     const REMOVE: bool = false;
-    let Some(attr) = field.attrs.iter().find(|attr| {
-        attr.path().segments.len() == 1 && attr.path().segments.first().unwrap().ident == "characteristic"
-    }) else {
+
+    let Some(attr) = field.attrs.iter().find(|attr| attr.path().is_ident("characteristic")) else {
         return RETAIN; // If the field does not have a characteristic attribute, retain it.
     };
-    let args = match CharacteristicArgs::parse(attr) {
+    let mut descriptors = Vec::new();
+    let mut doc_string = String::new();
+    let mut characteristic_count = 0;
+    for attr in &field.attrs {
+        if let Some(ident) = attr.path().get_ident() {
+            match ident.to_string().as_str() {
+                "doc" => {
+                    if let Ok(meta_name_value) = attr.meta.require_name_value() {
+                        if let syn::Expr::Lit(value) = &meta_name_value.value {
+                            if let Some(text) = &value.lit.span().source_text() {
+                                let text: Vec<&str> = text.split("///").collect();
+                                if let Some(text) = text.get(1) {
```

Comment:
So you're starting with a doc string line, you split at `///` to get anything before the '///' as text[0] and anything after as text[1]?
Would this filter out "blank" docstring lines, as in 
```rust
///
```
and if so is that what we want to do?

Reply:
Good question, we do not want to filter them out.

Reply:
Ok I've tried it and it does still allow empty lines with this code. It would just deny an 'initial' empty line.